ResearchKit/Common/ORKLocationSelectionView.m

```diff
@@ -0,0 +1,343 @@
+    [self addSubview:_mapView];
+    
+    if (_answer) {
+        CLLocationCoordinate2D coordinate = [((NSValue *)_answer) MKCoordinateValue];
+        CLLocation *location = [[CLLocation alloc] initWithLatitude:coordinate.latitude longitude:coordinate.longitude];
+        [self addAnnotationForLocation:location];
+    }
+    
+    if (!animated) {
+        [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|[_mapView]|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:nil views:NSDictionaryOfVariableBindings(_mapView)]];
+        [self removeConstraint:_textFieldBottomConstraint];
+        _textFieldBottomConstraint = [NSLayoutConstraint constraintWithItem:_textField attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:_mapView attribute:NSLayoutAttributeTop multiplier:1.0 constant:0.0];
+        [self addConstraint:_textFieldBottomConstraint];
+        _mapViewHeightConstraint = [NSLayoutConstraint constraintWithItem:_mapView attribute:NSLayoutAttributeHeight relatedBy:NSLayoutRelationEqual toItem:nil attribute:NSLayoutAttributeNotAnAttribute multiplier:1.0 constant:228.0];
+        [self addConstraint:_mapViewHeightConstraint];
+        [self addConstraint:[NSLayoutConstraint constraintWithItem:_mapView attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeBottom multiplier:1.0 constant:0.0]];
+        [self layoutIfNeeded];
+    } else {
+        [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|[_mapView]|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:nil views:NSDictionaryOfVariableBindings(_mapView)]];
+        [self removeConstraint:_textFieldBottomConstraint];
+        _textFieldBottomConstraint = [NSLayoutConstraint constraintWithItem:_textField attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:_mapView attribute:NSLayoutAttributeTop multiplier:1.0 constant:0.0];
+        [self addConstraint:_textFieldBottomConstraint];
+        _mapViewHeightConstraint = [NSLayoutConstraint constraintWithItem:_mapView attribute:NSLayoutAttributeHeight relatedBy:NSLayoutRelationEqual toItem:nil attribute:NSLayoutAttributeNotAnAttribute multiplier:1.0 constant:0.0];
+        [self addConstraint:_mapViewHeightConstraint];
+        [self addConstraint:[NSLayoutConstraint constraintWithItem:_mapView attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeBottom multiplier:1.0 constant:0.0]];
+        
+        [self layoutIfNeeded];
+        _mapViewHeightConstraint.constant = 228.0;
+        __weak __typeof__(self) weakSelf = self;
+        [UIView animateWithDuration:0.3 animations:^{
+            [weakSelf layoutIfNeeded];
+            
+            if ([weakSelf.delegate respondsToSelector:@selector(selectionViewNeedsResize:)]) {
+                [weakSelf.delegate selectionViewNeedsResize:self];
+            }
+        }];
```

Comment:
Create a strong reference to the weakSelf inside the block because the weakSelf is at risk of being deallocated during the execution of the block.